Client/mods/deathmatch/logic/luadefs/CLuaEngineDefs.cpp

```diff
@@ -1317,3 +1329,486 @@ int CLuaEngineDefs::EngineResetSurfaceProperties(lua_State* luaVM)
 
     return 1;
 }
+
+int CLuaEngineDefs::EngineGetModelPhysicalPropertiesGroup(lua_State* luaVM)
+{
+    //  int engineGetModelPhysicalPropertiesGroup ( int modelID )
+    int iModelID;
+
+    CScriptArgReader argStream(luaVM);
+    argStream.ReadNumber(iModelID);
+
+    if (!argStream.HasErrors())
+    {
+        if (iModelID < 0 || iModelID >= 20000)
+        {
+            argStream.SetCustomError("Expected model ID in range [0-19999] at argument 1");
+            m_pScriptDebugging->LogCustom(luaVM, argStream.GetFullErrorMessage());
+            lua_pushnil(luaVM);
+            return 1;
+        }
+
+        auto pModelInfo = g_pGame->GetModelInfo(iModelID);
+        if (pModelInfo)
+        {
+            lua_pushnumber(luaVM, pModelInfo->GetObjectPropertiesGroup());
+            return 1;
+        }
+        argStream.SetCustomError("Expected valid model ID at argument 1");
+    }
+    if (argStream.HasErrors())
+        m_pScriptDebugging->LogCustom(luaVM, argStream.GetFullErrorMessage());
+
+    // We failed
+    lua_pushnil(luaVM);
+    return 1;
+}
+
+int CLuaEngineDefs::EngineSetModelPhysicalPropertiesGroup(lua_State* luaVM)
+{
+    //  bool engineSetModelPhysicalPropertiesGroup ( int modelID, int newGroup )
+    int          iModelID;
+    unsigned int iNewGroup;
+
+    CScriptArgReader argStream(luaVM);
+    argStream.ReadNumber(iModelID);
+    argStream.ReadNumber(iNewGroup);
+
+    if (!argStream.HasErrors())
+    {
+        if (iModelID < 0 || iModelID >= 20000)
+        {
+            argStream.SetCustomError("Expected model ID in range [0-19999] at argument 1");
+            m_pScriptDebugging->LogCustom(luaVM, argStream.GetFullErrorMessage());
+            lua_pushnil(luaVM);
+            return 1;
+        }
+
+        if (iNewGroup < 0 || iNewGroup > 159)
+        {
+            argStream.SetCustomError("Expected group ID in range [0-159] at argument 1");
```

Comment:
doesn't quite line up with the error above (0 - 19999 includes 19999, 0 - 159 excludes 159?

Reply:
Changed the upper check to match the format, and now it check against 19999